src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/CustomEventListenerProvider.java

```diff
@@ -0,0 +1,72 @@
+package com.defenseunicorns.uds.keycloak.plugin;
+
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.keycloak.events.Event;
+import org.keycloak.events.EventListenerProvider;
+import org.keycloak.events.EventType;
+import org.keycloak.events.admin.AdminEvent;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.RealmProvider;
+import org.keycloak.models.UserModel;
+
+public class CustomEventListenerProvider implements EventListenerProvider {
+
+    private final KeycloakSession session;
+    private final RealmProvider model;
+
+    public CustomEventListenerProvider(KeycloakSession session) {
+        this.session = session;
+        this.model = session.realms();
+    }
+
+    @Override
+    public void onEvent(Event event) {
+
+        if (EventType.REGISTER.equals(event.getType())) {
+
+            RealmModel realm = this.model.getRealm(event.getRealmId());
+            UserModel newRegisteredUser = this.session.users().getUserById(realm, event.getUserId());
+
+            generateMattermostId(newRegisteredUser);
```

Comment:
Question: Is there any need in here to check if the user already has a mattermostId before generating a new one? Since the id is date-based, if we were to ever regenerate it would change.

Reply:
I think since we are only listening for the `REGISTER` event we should be safe? But it might not hurt to add a check regardless... curious on @UnicornChance's thoughts.